### PR TITLE
feat(reports): RHICOMPL-1236 check download URL

### PR DESCRIPTION
**Pull request by vkrizan**
Check schema of the report download URL to be https/secured in
production.  Fail otherwise.

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [X] Input Validation
- [X] Output Encoding
- [X] Authentication and Password Management
- [X] Session Management
- [X] Access Control
- [X] Cryptographic Practices
- [X] Error Handling and Logging
- [X] Data Protection
- [X] Communication Security
- [X] System Configuration
- [X] Database Security
- [X] File Management
- [X] Memory Management
- [X] General Coding Practices

### CODE REVIEW

File: app/services/safe_downloader.rb [hunk 1]
```diff
@@ -55,11 +55,18 @@ def encode_url(url)
...
+      check_url(url)
       url
     rescue ArgumentError
       raise DownloadError, 'url was invalid'
     end
 
+    def check_url(url)
+      return unless Rails.env.production?
+
+      raise DownloadError, 'not secured (not https)' if url.scheme != 'https'
```
Comment (skateman) [inline app/services/safe_downloader.rb:67]:
```suggestion
      raise DownloadError, 'not secure (non-https)' if url.scheme != 'https'
```

File: app/services/safe_downloader.rb [hunk 2]
```diff
@@ -55,11 +55,18 @@ def encode_url(url)
       url = URI(url)
       raise DownloadError, 'url was invalid' unless url.respond_to?(:open)
 
+      check_url(url)
```
Comment (skateman) [inline app/services/safe_downloader.rb:58]: I'm wondering if we could "flatten out" the exception flow here, you're introducing a new one a level deeper and it makes the code a bit less followable :thinking:

**Review by skateman — APPROVED** [review]
Small nits only

File: app/services/safe_downloader.rb (hunk 2 above)
Comment (vkrizan) [reply app/services/safe_downloader.rb:58]: You mean the rerise of the same type with additional info? I've kinda used the flow that is used there already.

Comment (skateman) [reply app/services/safe_downloader.rb:58]: It's just weird that you're raising an error similarly to the previous one, just a lever deeper. But keep it so if rubocop forced you to extract this to a method :shrug:

Comment (vkrizan) [reply app/services/safe_downloader.rb:58]: Oh, I see what you mean.

**Pull request merged**